app/models/annotation_merger.rb

```diff
@@ -0,0 +1,101 @@
+class AnnotationMerger
+  def initialize(annotations)
+    @annotations = annotations
+    @chunks_info = build_chunks_info
+    @id_mappings = build_id_mappings
+  end
+
+  def merged
+    result = { "text" => merged_text }
+    result["denotations"] = merged_denotations if merged_denotations.any?
+    result["relations"] = merged_relations if merged_relations.any?
+    result
+  end
+
+  private
+
+  # Pre-calculate information for each chunk (length and whether it has padding)
+  def build_chunks_info
+    @annotations.each_with_index.each_with_object([]) do |(annotation, index), chunks_info|
```

Comment:
```suggestion
    @annotations.with_index.each_with_object([]) do |(annotation, index), chunks_info|
```
と書けるようです。

Reply:
<img width="729" height="198" alt="image" src="https://github.com/user-attachments/assets/03a0235c-d103-41cf-a199-994d4b1e60ca" />
エラーになります。
```
NoMethodError: undefined method 'with_index' for an instance of Array
    app/models/annotation_merger.rb:19:in 'AnnotationMerger#build_chunks_info'
    app/models/annotation_merger.rb:4:in 'AnnotationMerger#initialize'
```

Reply:
失礼しました。

```suggestion
    @annotations.each_with_object([]).with_index do |(annotation, chunk_info), index|
```

なら行けそうです。